test/testfolderstatusdelegate.cpp

```diff
@@ -0,0 +1,27 @@
+/*
+ *    This software is in the public domain, furnished "as is", without technical
+ *    support, and with no warranty, express or implied, as to its usefulness for
+ *    any purpose.
+ *
+ */
+
+#include <QtTest>
+#include <QPainter>
+
+#include "folderstatusmodel.h"
+#define private public
+#include "folderstatusdelegate.h"
+#undef private
```

Comment:
this "trick" will work because we are using a static library
if we would have been using a shared library (DLL on windows) I think that would fail because the symbol mangling is different for private/public class members (see https://en.wikiversity.org/wiki/Visual_C%2B%2B_name_mangling#Complex_Type_(union,_struct,_class,_coclass,_cointerface) and https://labs.trolltech.com/blogs/2009/08/12/some-thoughts-on-binary-compatibility/)
I would not block this PR only based on that but

- we would be unable to keep using that if our code was moved to a shared library (DLL)
- I am not convinced of automated tests using private interfaces
- That creates heavy coupling of the tests to the current internal implementation

why not "just" make your test a friend of the class you are testing ? That would also be more in line with the current state of the automated tests of the desktop client ?

Reply:
Yes, friend class is very simple and easy solution, I avoided this because of only data/member privacy concern.
but if this is not major concern then I will apply the same.